### PR TITLE
Add new "Design Index" endpoint for thawing design index and freezing design index

### LINKED ISSUE #190: Implement design thawing index and design freezing index endpoints

### LINKED ISSUE #192: API does not parse Rasdaman coverage metadata with multiple dimensions properly
For API endpoints that read their coverage metadata from Rasdman, coverages with multiple dimensions do not seem to parse properly and will throw an exception.

For example, the API will read the following Rasdaman coverage metadata for `thawing_index` properly and store it in the `dim_encodings` variable:

```
<Encoding>{'model': {'1': 'GFDL-CM3', '0': 'ERA-Interim', '2': 'NCAR-CCSM4'}}</Encoding>
```

But the following coverage metadata for `design_thawing_index` does not parse properly:

```
<Encoding>{'model': {'1': 'GFDL-CM3', '0': 'ERA-Interim', '2': 'NCAR-CCSM4'}, 'era': {'0': '1980-2009', '1': '2040-2069', '2': '2070-2099'}}</Encoding>
```

### PR DESCRIPTION
**Pull request by cstephen**
Closes #190.
Closes #192.

This PR adds a new "Design Index" endpoint to the `degree_days.py` file that serves data from the following Rasdaman coverages:

- design_thawing_index
- design_freezing_index

Unlike the four other degree days endpoints, which are served as subpaths of the `/mmm` URL path, the Design Index endpoint is accessed through a new `/design_index` URL path because it does not serve the same kind of min/mean/max data that the endpoints in `/mmm` do.

To test, make sure your local API is pointed at Apollo:

```
export API_RAS_BASE_URL=http://apollo.snap.uaf.edu:8080/rasdaman/
```

Then make sure all of the URLs in the Design Index endpooint documentation work as expected, and experiment with other coordinates too.

This PR also includes some minor fixes to the Degree Days endpoint documentation that I missed in my previous PR, and a fix that allows the API to parse multi-dimension Rasdaman coverage metadata properly, which was needed to parse the metadata for the `design_thawing_index` and `design_freezing_index` coverages (see issue #192 for details).

It looks like this PR picked up a lot of the commit history from the `degree_days` branch it was branched from. I think merging this using "Squash and merge" is our best bet to keep the `main` branch git history clean.

### CODE REVIEW

File: routes/degree_days.py [hunk 1]
```diff
@@ -252,52 +301,80 @@ async def fetch_dd_point_data(x, y, cov_id, horp, start_year, end_year):
...
+            "variable",
+            "value",
+        ]
+    else:
+        fieldnames = [
+            "model",
+            "year",
+            "variable",
+            "value",
+        ]
```
Comment (kyleredilla) [inline routes/degree_days.py:345]: Do we need this change? It looks like `fieldnames` is the same in both cases of the `if` statement?

**Review by kyleredilla — CHANGES_REQUESTED** [review]
These changes look good. I've tested all of the examples from the new endpoints in the docs with various coordinate combinations, and I've also tested some samples from the degree_days endpoint as well. Things worked as expected. The additional tweaks to existing code are good findings to incorporate as well. I approve this PR for merging after my single question is addressed.

File: routes/degree_days.py (hunk 1 above)
Comment (cstephen) [reply routes/degree_days.py:345]: Good catch! Somewhere along the way while I was working on this the fieldnames became the same for degree days vs. design index. I'll get this fixed.

Comment (cstephen) [reply routes/degree_days.py]: Fixed in 160f8ec63c33670c5f7c329fe9a4086e55bed377.

**Review by kyleredilla — APPROVED** [review]
Modifications look good, I approve!

**Pull request merged**